src/bitcoin-cli.cpp

```diff
@@ -191,7 +192,9 @@ static void http_error_cb(enum evhttp_request_error err, void *ctx)
 
 UniValue CallRPC(const std::string& strMethod, const UniValue& params)
 {
-    std::string host = GetArg("-rpcconnect", DEFAULT_RPCCONNECT);
+    std::string host;
+    int dummy_port;
+    SplitHostPort(GetArg("-rpcconnect", DEFAULT_RPCCONNECT), dummy_port, host);
```

Comment:
I'd really prefer to be able to override the port in rpcconnect.
FYI we have a similar thing at the server side with `bind` which can either take a address and port (overriding `-port`) or just an address (in which case `-port` is used).
If not I'd rather NACK this. It is just confusing to parse an `[ip]:port` spec if you're going to ignore the port.